splunk_otel/profiling/__init__.py

```diff
@@ -0,0 +1,300 @@
+# Copyright Splunk Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import base64
+import gzip
+import logging
+import os
+import sys
+import threading
+import time
+import traceback
+from collections import OrderedDict
+from typing import Dict, Optional, Union
+
+import wrapt
+from opentelemetry._logs import SeverityNumber
+from opentelemetry.context import Context
+from opentelemetry.exporter.otlp.proto.grpc._log_exporter import OTLPLogExporter
+from opentelemetry.sdk._logs import LogData, LogRecord
+from opentelemetry.sdk._logs.export import BatchLogRecordProcessor
+from opentelemetry.sdk.util.instrumentation import InstrumentationScope
+from opentelemetry.semconv.resource import ResourceAttributes
+from opentelemetry.trace import TraceFlags
+from opentelemetry.trace.propagation import _SPAN_KEY
+
+import splunk_otel
+from splunk_otel.profiling import profile_pb2
+from splunk_otel.profiling.options import _Options
+from splunk_otel.version import __version__
+
+logger = logging.getLogger(__name__)
+
+
+class Profiler:
```

Comment:
In this case I wanted to just store the globals somewhere (batch processor, running flag), not sure hiding behind these methods will be more useful, or will it?

Reply:
I do think encapsulating a class's fields behind methods makes the class easier to understand and maintain because the amount of API exposed becomes smaller, and then callers don't have to know about the class's internals.